include/gatekeeper_acl.h

```diff
@@ -60,20 +60,25 @@ add_pkt_acl(struct acl_search *acl, struct rte_mbuf *pkt)
 	acl->num++;
 }
 
-static inline int
-ipv4_acl_enabled(struct gatekeeper_if *iface)
+static inline bool
+ipv4_acl_enabled(const struct gatekeeper_if *iface)
 {
 	/*
-	 * The IPv4 ACL is only needed for interfaces that have
-	 * IPv4 addresses and don't support the ntuple filter.
+	 * The IPv4 ACL is needed whenever an interface has an IPv4 address.
+	 * Even if ntuple filters are available, the IPv4 ACL is always
+	 * needed to be able to match ICMP packets.
```

Comment:
This function is making assumptions on the behavior of the blocks and hardware availability; these assumptions are troubling when we enhance this interface down the line, what we now know is likely to happen. The way to avoid these assumptions is to add the field `bool acl_ipv4_enabled` in `struct gatekeeper_if`. This patch should add this new field, but it doesn't need to set it since `ipv4_pkt_filter_add()` in the following patch is the proper place to do it.

This comment also applies to `ipv6_acl_enabled()`.

Reply:
Shouldn't this new field be set at initialization, instead of `ipv4_pkt_filter_add()` (which is called multiple times)?

Reply:
`struct gatekeeper_if` is zeroed for both front and back interfaces during initialization (isn't it?), and only the multiple calls of `ipv4_pkt_filter_add()` can make field `acl_ipv4_enabled` true. Once Gatekeeper enjoys more hardware support, `ipv4_pkt_filter_add()` may never set field `acl_ipv4_enabled` true, so we cannot initialize it before all the calls to `ipv4_pkt_filter_add()`.

Adding a comment to field `acl_ipv4_enabled` explaining how it's set is helpful.

Reply:
The employed alternative solution introduced a bug. See newer review for details.